docs: proofread market terminology and fronted pages (#70)
NilFoundation/proof-market-toolchain: gitbook-docs/market/front-end.md

---
description: Web interface of the Proof Market
---

# Web interface

> The web interface of the Proof Market can be accessed at [**proof.market**](https://proof.market).

Reviewer: Link formatting is enough to highlight the link from the surrounding text, so there's no need for bold.
```suggestion
The web interface of the Proof Market can be accessed at [proof.market](https://proof.market).
```

Author: Fixed

{% hint style="info" %}
Please note this product is in beta and expected to change.
{% endhint %}

## Creating an account 

> To interact with the Proof Market, create an account at [proof.market](https://proof.market)

Reviewer: This link can lead directly to the Sign up page.

or [through the command-line interface](cmd-reference/user.md).
Without an account, you can view the exchange, but you won't be able to post orders 
to buy or sell proofs.

To register a new user via the web interface, click **Sign in** from the main page
and then click **Sign up**:

<figure><img src="../.gitbook/assets/sign-in.png" alt=""><figcaption><p>Sign in view</p></figcaption></figure>

> Or you can go directly to the [register page](https://proof.market/#/register).

Reviewer: This is easier than clicking through the "Sign in" screen. Let's put it before the picture.

To make an account on the Proof Market, go to the [Sign up](https://proof.market/#/register) page or select **Sign in** > **Sign up** on the main page.

Enter your username and password and complete the flow by clicking on **Register**.

<figure><img src="../.gitbook/assets/sign-up.png" alt=""><figcaption><p>Registration view</p></figcaption></figure>

## Logging in

If you're a registered user, enter your credentials in the provided fields and click **Login**.

<figure><img src="../.gitbook/assets/login.png" alt=""><figcaption><p>Login view</p></figcaption></figure>

## Dashboard

The Proof Market dashboard has the following components:

* [A: Circuit list](front-end.md#circuit-list);
* [B: Trades](front-end.md#trades);
* [C: Last proof producer](front-end.md#last-proof-producer);
* [D: Circuit details](front-end.md#circuit-details);
* [E: Circuit price dashboard](front-end.md#circuit-price-dashboard);
* [F: Order book](front-end.md#order-book);
* [G: Create orders](front-end.md#create-orders);
* [H: Manage orders](front-end.md#manage-orders).

<figure><img src="../.gitbook/assets/dashboard.png" alt=""><figcaption><p>The Proof Market dashboard</p></figcaption></figure>

> ### Circuit list

Author: This calls for another task. The screenshots are outdated, and thus so is description. @NickVolynkin, do we have a task for it already?

> ### Circuit list

Author: Thanks, fixed. The screenshots are also to be updated

> The circuit/statements list shows the instruments available on the Proof Market

Reviewer: Let's use the term "circuit statements" at the first appearance on a page, and then we can reduce it to "statements."

> The circuit/statements list shows the instruments available on the Proof Market

Author: Instrument/circuit/statement terminology is explained on the Terminology page. I've added a link to it here

and their price change in the last 24 hours.
Here you can search the instruments to create an order to buy or sell proofs.
Selection of an instrument in this view also affects all the other components.

<figure><img src="../.gitbook/assets/circuit-list.png" alt=""><figcaption><p>Circuit list view</p></figcaption></figure>

### Trades

> The trades component shows all the orders executed on the marketplace for the selected circuit.

Author: Thanks, fixed

<figure><img src="../.gitbook/assets/trades.png" alt=""><figcaption><p>Trades view</p></figcaption></figure>

### Last proof producer

This section shows the username of the last proof producer on the marketplace.

<figure><img src="../.gitbook/assets/last-proof-producer.png" alt=""><figcaption><p>Last producer view</p></figcaption></figure>

### Circuit details

> The circuit details component shows more information about the circuit and its aggregated

Author: Thanks, fixed

marketplace stats.

<figure><img src="../.gitbook/assets/circuit-details.png" alt=""><figcaption><p>Circuit details view</p></figcaption></figure>

> ### Circuit price dashboard

Reviewer: And we don't sell circuits, we sell proofs.

> ### Circuit price dashboard

Author: How about "Proof dashboard"?

> ### Circuit price dashboard

Reviewer 2: I think, "Statement dashboard" should be fine. We don't sell them but the information about the particular statement

This comprises two views:

> * Proof cost — a history of the price settlement for this circuit.

Author: @SK0M0R0H Volume as in market volume, amount of transactions?

> * Proof cost — a history of the price settlement for this circuit.

Reviewer 2: Amount of generated proofs, at the bottom of the chart with prices

> * Proof cost — a history of the price settlement for this circuit.

Author: Fixed

<figure><img src="../.gitbook/assets/circuit-dashboard-cost.png" alt=""><figcaption><p>Proof cost</p></figcaption></figure>

* Proof generation time — a history of how long did proof generation take from the time

> the order was matched to the time a proof was submitted to the market.

Reviewer: remember the indentation
```suggestion
  the order was matched to the time a proof was submitted to the market.
```

<figure><img src="../.gitbook/assets/circuit-dashboard-gen-time.png" alt=""><figcaption><p>Proof generation time</p></figcaption></figure>

### Order book

The order book shows the request and proposals on the marketplace for orders
which have not yet matched.

<figure><img src="../.gitbook/assets/order-book.png" alt=""><figcaption><p>Order book view</p></figcaption></figure>

You can toggle to see your orders in the book and group them.

### Create orders

You can learn more about orders and their fields in [Terminology](overview.md/#orders-types).

> Please note that you should add your circuit as a statement in JSON format

Author: My bad, didn't commit that change. Here it is

---
description: Web interface of the Proof Market
---

# Web interface

The web interface of the Proof Market can be accessed at [**proof.market**](https://proof.market).

{% hint style="info" %}
Please note this product is in beta and expected to change.
{% endhint %}

## Creating an account 

To interact with the Proof Market, create an account at [proof.market](https://proof.market)
or [through the command-line interface](cmd-reference/user.md).
Without an account, you can view the exchange, but you won't be able to post orders 
to buy or sell proofs.

> To register a new user via the web interface, click **Sign up**:

Author: Please check this out, I've updated this line according to the current design of proof.market

> To register a new user via the web interface, click **Sign in** and then **Sign up**:

Author: @NickVolynkin Please check this out, I've updated this line according to the current design of proof.market

Reviewer: Let's give a direct link as well: https://proof.market/#/register

Author: Done, but I've placed it a couple of lines below this, since the first screenshot is about the main page

---

> description: The web interface of the Proof Market

Reviewer: I have a feeling that we can omit 'the' in headings.

Author: fixed

---

# Web interface

{% hint style="info" %}
Please note this product is in beta and expected to change.
{% endhint %}

> #### The web interface of the Proof Market can be accessed at [**proof.market**](https://proof.market).

Reviewer: It's not a good text for a heading: it's a complete sentence with a link. So it should be just a paragraph.

Author: fixed

> ## New user signup

Reviewer: First, only new users sign up, the non-new ones log in. Second, let's give it a better name.

```suggestion
## Creating an account
```

Author: fixed

> All users who wish to interact with the Proof Market must create an account.

Reviewer: Can be said much simpler.

To interact with the Proof Market, [create an account](link here).
Without an account you can view the exchange, but you need an account to post orders to buy or sell proofs.

And it doesn't look like a callout. It's the main topic of this section, while a callout is something out of the main narrative.

Author: fixed

{% hint style="info" %}
If you do not have an account, you can still view the exchange, but you won't be able to post orders
to buy or sell proofs.
{% endhint %}

> To register a new user via the web interface, click on **Sign up**:

Reviewer: https://developers.google.com/style/ui-elements#term-menus

```suggestion
To register a new user via the web interface, click **Sign up**:
```

Author: fixed

> <figure><img src="../.gitbook/assets/image (6).png" alt=""><figcaption><p>Sign up view</p></figcaption></figure>

Reviewer: Could you please fix the file names as well? They should have clear names without spaces and other punctuation.

```diff
- image (6).png
+ sign-up.png

- image.png
+ registration-form.png
```

Author: fixed

Enter your username and password and complete the flow by clicking on **Register**.

<figure><img src="../.gitbook/assets/image.png" alt=""><figcaption><p>Registration view</p></figcaption></figure>

> ## Login

Reviewer: https://grammarist.com/spelling/log-in-login/

Let's say "Logging in"

If you're a registered user, enter your credentials in the provided fields and click **Login**.

> <figure><img src="../.gitbook/assets/image (7).png" alt=""><figcaption><p>Login view</p></figcaption></figure>

Reviewer: Here you've put the elements in the right order: heading, text, image. It's much better than when there's an image right after the heading.

Please fix other sections in the same way.

Author: fixed

## Dashboard

<figure><img src="../.gitbook/assets/image (11).png" alt=""><figcaption><p>The Proof Market dashboard</p></figcaption></figure>

> The Proof Market dashboard can be broken down into the following components:

Reviewer: Somewhy I don't like this "broken down" used with our service that we actually don't want to break down. Let's say that it has the following components.

* [A: Circuit list](front-end.md#circuit-list);
* [B: Trades](front-end.md#trades);
* [C: Last proof producer](front-end.md#last-proof-producer);
* [D: Circuit details](front-end.md#circuit-details);
* [E: Circuit price dashboard](front-end.md#circuit-price-dashboard);
* [F: Order book](front-end.md#order-book);
* [G: Create orders](front-end.md#create-orders);

> * [H: Manage orders](front-end.md#manage-orders).

Reviewer: Are these letters used anywhere else? If not, we don't need them here.

Author: Yep, zones of the UI are marked on a screenshot with those letters

Reviewer: Oh, fine then.

This section shows the username of the last proof producer on the marketplace.

### Circuit details

<figure><img src="../.gitbook/assets/image (5).png" alt=""><figcaption><p>Circuit details view</p></figcaption></figure>

The circuit details component shows more information about the circuit and its aggregated
marketplace stats.

### Circuit price dashboard

This comprises two views:

* Proof cost — a history of the price settlement for this circuit.

<figure><img src="../.gitbook/assets/image (4) (1).png" alt=""><figcaption><p>Proof cost</p></figcaption></figure>

* Proof generation time — a history of how long did proof generation take from the time
the order was matched to the time a proof was submitted to the market.

<figure><img src="../.gitbook/assets/image (6) (1).png" alt=""><figcaption><p>Proof generation time</p></figcaption></figure>

### Order book

The order book shows the request and proposals on the marketplace for orders which have not yet matched.

<figure><img src="../.gitbook/assets/image (4).png" alt=""><figcaption><p>Order book view</p></figcaption></figure>

You can toggle to see your orders in the book and group them.

### Create orders

> Create orders section can be used to place orders:

Author: fixed

* **Buy order**: A buy order is posted by a proof requester.
  Each buy order comprises the following parameters:

>   * **Cost** — the amount in USD that the user is willing to pay for the proof.

Reviewer: Sometimes we speak directly to the reader: _If you're a registered user, enter your credentials_
But at other times, there's this _user_, who is a third party to us and our readers. We should discuss it later.

Let's rename the user to a proof requester, at least.

Author: fixed

>   * **Cost** — the amount in USD that the user is willing to pay for the proof.
>   * **Order time out** — time in minutes; how long should the order stay in the order book
>     once it enters the marketplace.
>     If it times out before being matched, the order will be purged.
>   * **Generation time** — the maximum time the proof generator has to publish the proof once the order
>     is matched.
>     This is an optional parameter expressed in minutes.
>   * **Public inputs** — an optional parameter to add public inputs if the circuit requires it.

Reviewer: Here we have a typical situation: there are input variables, and we should describe their 1) meaning and 2) data type.

It's best to do it consistently between all variables.
```suggestion
  * **Cost** (in USD): the price that the user is willing to pay for the proof.
  * **Order timeout** (in minutes): how long an order stays in the order book
    once it enters the marketplace.
    If it times out before being matched, the order will be purged.
  * **Generation time** (in minutes, optional): the maximum time the proof generator has to publish the proof
  once the order is matched.
  * **Public inputs** (optional): public input data if the circuit requires it.
```

Author: fixed

Author: > Maybe we should only describe order types on the Terminology page, link it from the Frontend
Done that
